Content.Shared/Weapons/Reflect/ReflectSystem.cs

```diff
@@ -91,16 +101,43 @@ private void OnReflectCollide(EntityUid uid, ReflectComponent component, ref Pro
 
     private bool TryReflectProjectile(EntityUid user, EntityUid reflector, EntityUid projectile, ProjectileComponent? projectileComp = null, ReflectComponent? reflect = null)
     {
+        // Do we have the components needed to try a reflect at all?
         if (!Resolve(reflector, ref reflect, false) ||
             !reflect.Enabled ||
             !TryComp<ReflectiveComponent>(projectile, out var reflective) ||
             (reflect.Reflects & reflective.Reflective) == 0x0 ||
-            !_random.Prob(reflect.ReflectProb) ||
             !TryComp<PhysicsComponent>(projectile, out var physics))
-        {
             return false;
+
+        var reflectChance = reflect.ReflectProb;
+
+        /*
+         *  The rules of deflection are as follows:
+         *  If you innately reflect things via magic, biology etc, you always have a full chance.
+         *  If you are standing up and standing still, you're prepared to deflect and have full chance.
+         *  If you are moving at all, your chance is multiplied by the movingProbMultiplier.
+         *  If you are recklessly sprinting, your chance is instead multiplied by the sprintingProbMultiplier.
+         *  If you floating, your chance is instead multiplied by the sprintingProbMultiplier.
+         *  You cannot deflect if you are knocked down or stunned.
+         */
+
+        if (!reflect.Innate)
+        {
+            if (TryComp<StaminaComponent>(reflector, out var staminaComponent) && staminaComponent.Critical)
+                return false;
+
+            if (_standing.IsDown(reflector))
+                return false;
+
+            if (_gravity.IsWeightless(reflector))
+                reflectChance *= reflect.SprintingProbMultiplier;
+            else if (TryComp<InputMoverComponent>(reflector, out var mover) && (mover.HeldMoveButtons & MoveButtons.AnyDirection) == MoveButtons.AnyDirection)
```

Comment:
Not a required change (unless another maintainer wishes to declare it as such), but it'd probably behave more naturally for the multiplier to be based off velocity, as that'd allow the mechanic to work seamlessly with NPCs instead of needing to rely on the innate modifier, would allow the mechanic to work proper when weightless, would make it easy to avoid trivially cheesing the mechanic with low-friction, and would avoid requiring a future rewrite whenever analog support for movement is added.

Reply:
This ends up being a tad tricky to write in a way that allows for expression, though. If Urist McChonker is fleeing at high speed at 1 tile a second because they're weighed down by various factors, they are still deciding to flee rather than stand their ground.

I'm on the side of game feel here as it ties the deflection into inputs the user is making, so a higher-skill player can tune their movement (accounting for server lag) to be more effective.

Reply:
It'll Just Work, trust.

Base movement speeds, current movement speed (not to be confused with velocity), and velocity, are all separate values, and these values can be used in various ways to calculate whether or not an entity should be considered moving. Whether it's a threshold or a proper lerp (or exponential interp if you're feeling spicy), it'd simply Just Work for all intents and purposes while also avoiding quite a few forms of jank and reducing the potential for mechanical oversights. The option of using a lerp comes with the added benefit of the mechanic behaving far more naturally (as suffering from slowdown would give you a reflection probability right between whatever two states you're between. This makes sense from a gameplay standpoint, even if it does introduce more effective randomness to the table)

Counter-Strike is a good example of a game that does something similar (and even has *very* similar movement physics to boot!). In Counter-Strike, weapon inaccuracy is determined entirely by the velocity of your player movement, with any velocity faster than a given threshold (about half the walk speed in CSGO's case) causing the inaccuracy to lerp from 0 to its maximum value after a different threshold (which in CSGO's case is somewhere fairly close to the full run speed). This is a surprisingly graceful way of handling it, as it effectively communicates to other players whether or not your shots are going to be at full accuracy, while also enabling skill expression in the form of counter-strafing (wherein you cancel your movement out by tapping the opposite movement key, thereby allowing full accuracy to be achieved pretty quickly), and ensuring that players can't end up creatively violating the main design intent of needing to stay still for benefits.

Reply:
Good feedback well delivered. Will see what I can do tonight.